### ISSUE TITLE
RefinementList label customization

### ISSUE THREAD

**Issue by LeComptoirDesPharmacies** [algolia/react-instantsearch#3136]
**Is your feature request related to a problem? Please describe 🙏**

RefinementList component can not receive custom label like ToggleRefinement do. It always use raw 'attribute' as label.
Thus, it is more difficult to display a beautiful label in `CurrentRefinements` component/connector.

**Describe the solution you'd like 🤔**

Two possible solutions : 

1. Add a `label` prop to `connectCurrentRefinements` and `CurrentRefinements` and use it in `getMetadata` function.
2. Add a `translations` prop to `connectCurrentRefinements` and `CurrentRefinements` and use it to convert the attribute name to intelligible name.

**Describe alternatives you've considered ✨**

I am going to translate the label in my display component using attribute raw name (ex : owner.nickname) as source.

**Additional context**

No additional context.

**Comment by francoischalifour** [comment]
The `RefinementList` component supports [`transformItems`](https://www.algolia.com/doc/api-reference/widgets/refinement-list/react/#widget-param-transformitems) which allows you to modify the `label` before rendering it.

Does that help achieve what you're looking for?

**Comment by LeComptoirDesPharmacies** [comment]
Hi @francoischalifour,

In fact we are already using `connectRefinementList`. The `transformItems` is cool to customize label of items but here we would like to customize the main label of the `RefinementList` (Ex : 'owner.nickname' which is the attribute name).
If you check in the code of the connector (below) you can see that there is two level of label.
The item level is customizable through `transformItems` : 
```
items: getCurrentRefinement(props, searchState, context).map(
    item => ({
      label: `${item}`,
      {...}
```

And this main label is not : 
```
{
  attribute: props.attribute,
  label: `${props.attribute}: `,
  {...}
```

If you take a look at `connectToggleRefinement.js` you will see that we can customize the main label : 
```
 if (checked) {
      items.push({
        label: props.label,
        currentRefinement: checked,
        {...}
```

Yours faithfully,
LCDP

**Comment by Haroenv** [comment]
Do you mean in current refinements? You can also use transformItems inside currentRefinements, but the metadata can't be transformed by the widget itself. Does that work as a workaround @LeComptoirDesPharmacies ?

**Comment by callmekatootie** [comment]
I believe what they are requesting for is the ability to define a title for the entire RefinementList widget - If we have multiple such widgets then it is difficult to know which filter attribute is the widget for...

**Comment by Haroenv** [comment]
@callmekatootie, for that you can use the Panel widget: https://www.algolia.com/doc/api-reference/widgets/panel/react/

**Comment by LeComptoirDesPharmacies** [comment]
I am not sure that Panel widget will change the label provided in `connectCurrentRefinements` ?
Objective is to create beautiful pills which allow to clear some filters.

Problem is that `connectCurrentRefinements` give us the current refinements with their associated label.
These labels are customizable for some connectors (ex : connectToggleRefinement) but not for others (ex : connectRefinementList).

Current workaround we did is to do some Regexp Replace of labels given in `connectCurrentRefinements` to do beautiful display.
For example : 
`new RegExp('(.+) <= stock.remainingQuantity'` => `'Stock min.: $1'`

**Comment by Haroenv** [comment]
Yes, that's unfortunately the best option I see for now @LeComptoirDesPharmacies, I more think that @callmekatootie was asking a different question, not related to current refinements?

**Comment by callmekatootie** [comment]
Yep - I thought that's what @LeComptoirDesPharmacies was asking about too. I am mistaken though.

Anyhow - that definitely answered my question. I was looking for the Panel Widget indeed.
